core/storage/classifier/gae_models.py

```diff
@@ -174,3 +174,84 @@ def create(
 
         training_job_instance.put()
         return instance_id
+
+
+class ClassifierExplorationMappingModel(base_models.BaseModel):
+    """Model for mapping exploration attributes to a ClassifierDataModel.
+
+    The id of instances of this class has the form
+    {{exp_id}}.{{exp_version}}.{{state_name}}
+    """
+
+    # The ID of the classifier corresponding to the exploration attributes.
+    classifier_id = ndb.StringProperty(required=True, indexed=True)
+
+    @classmethod
+    def _generate_id(cls, exp_id, exp_version, state_name):
+        """Generates a unique ID for the Classifier Exploration Mapping of the
+        form {{exp_id}}.{{exp_version}}.{{state_name}}
+
+        Args:
+            exp_id: str. ID of the exploration.
+            exp_version: int. The exploration version at the time
+                this training job was created.
+            state_name: str. The name of the state to which the classifier
+                belongs.
+
+        Returns:
+            ID of the new Classifier Exploration Mapping instance.
+        """
+        if isinstance(state_name, unicode):
```

Comment:
Don't do it like this. Stuff that calls this should ensure the state name is unicode before passing it in. Here, just verify that the state name is indeed unicode, and then encode it.